### PR TITLE
refactor: let SearchIcon inherit color from style context

### PR DESCRIPTION
**Pull request by UncleBill**
also remove @masknet/theme from icons' dependencies

### CODE REVIEW

**Review by septs — APPROVED** [review]

**Comment by UncleBill** [comment]
I want to import an icon from `@masknet/icons` in `@masknet/theme`. `@/icons` already dependents `@/theme`, I don't want them to become a circular dependency to each other.

The reason I want an icon in `@/theme` is that a custom Snakbar I create in #3718 needs a spinning icon.

Reconsidering about the relationship between `@/theme` and `@/icons`, I think `@/theme` is a fundamental thing, it should be depended by any others, then we might need to move the components in `@/theme` to a new package to make `@/theme` more pure.

Would like to hear your opinion. @Jack-Works

**Review by septs — APPROVED** [review]

File: packages/icons/package.json [hunk 1]
```diff
@@ -9,9 +9,6 @@
     "start": "snowpack dev",
     "build": "node --es-module-specifier-resolution=node ./dist/utils/ssr.js"
   },
-  "dependencies": {
-    "@masknet/theme": "workspace:*"
```
Comment (Jack-Works) [inline packages/icons/package.json:13]: If you remove this, please also remove it from tsconfig file.

Comment (UncleBill) [reply packages/icons/package.json:13]: I don't see any reference to `@masknet/theme` in `packages/icon/tsconfig.json`. What I miss?

**Review by guanbinrui — APPROVED** [review]

File: packages/icons/package.json (hunk 1 above)
Comment (Jack-Works) [reply packages/icons/package.json:13]: Oh so that was a mistake when adding that dependency

**Pull request merged**